backend/src/relay.ts

```diff
@@ -224,6 +204,30 @@ export default class Relay {
     }
   }
 
+  private async getBlockWithJustification(rawBlock: SignedBlockJsonRpc) {
+    // adding justifications from finality proof if there is none
+    if (!rawBlock.justifications) {
+      const { number } = rawBlock.block.header;
+      const proof = await pRetry(
+        () => this.sourceApi.rpc.grandpa.proveFinality(number),
+        createRetryOptions(error => this.logger.error(error, `get block justifications for #${number} retry error:`)),
+      );
+      const proofBytes = proof.toU8a(true);
+      // finality proof returns encoded block hash, justification and unknown headers
+      // only care about justification, removing leading extra bytes: 32 for block hash and 4 for SCALE prefix(?)
```

Comment:
Number of bytes for scale codec should ideally be determined dynamically and not assumed that it is 4 bytes. You can just decode it as a vector and read it properly that way.